bundle/bundle.go

```diff
@@ -161,14 +140,74 @@ func Read(r io.Reader) (Bundle, error) {
 
 		} else if strings.HasSuffix(path, manifestExt) {
 			if err := util.NewJSONDecoder(&buf).Decode(&bundle.Manifest); err != nil {
-				return bundle, errors.Wrapf(err, "bundle load failed on manifest")
+				return bundle, errors.Wrap(err, "bundle load failed on manifest decode")
+			}
+
+			if r.includeManifestInData {
+				var metadata map[string]interface{}
+				b, err := json.Marshal(&bundle.Manifest)
+				if err != nil {
+					return bundle, errors.Wrap(err, "bundle load failed on manifest marshal")
+				}
+
+				err = util.UnmarshalJSON(b, &metadata)
+				if err != nil {
+					return bundle, errors.Wrap(err, "bundle load failed on manifest unmarshal")
+				}
```

Comment:
Something that is not entirely clear to me is, why do you go through the process of passing bundle.Manifest to and from json, in order to insert it into the bundle? why not directly pass bundle.Manifest to bundle.insert?